keps/sig-windows/20190103-windows-node-support.md

```diff
@@ -247,6 +258,102 @@ These areas still need test cases written:
+
+Huge pages are not implemented in the Windows container runtime, and are not available. They require [asserting a user privilege](https://docs.microsoft.com/en-us/windows/desktop/Memory/large-page-support) that's not configurable for containers.
+
+`V1.Container.ResourceRequirements.requests.cpu`
+`V1.Container.ResourceRequirements.requests.memory`
+
+Requests are subtracted from node available resources, so they can be used to avoid overprovisioning a node. However, they cannot be used to guarantee resources in an overprovisioned node. They should be applied to all containers as a best practice if the operator wants to avoid overprovisioning entirely.
+
+`V1.Container.SecurityContext.allowPrivilegeEscalation` - not possible on Windows, none of the capabilies are hooked up
+`V1.Container.SecurityContext.Capabilities` - POSIX capabilities are not implemented on Windows
+`V1.Container.SecurityContext.privileged` - Windows doesn't support privileged containers
+`V1.Container.SecurityContext.readOnlyRootFilesystem` - not possible on Windows, write access is required for registry & system processes to run inside the container
+`V1.Container.SecurityContext.runAsGroup` - not possible on Windows, no GID support
+`V1.Container.SecurityContext.runAsUser` - not possible on Windows, no UID support as int. This needs to change to IntStr, see [64009](https://github.com/kubernetes/kubernetes/pull/64009), to support Windows users as strings, or another field is needed.
+`V1.Container.SecurityContext.seLinuxOptions` - not possible on Windows, no SELinux
+
+`V1.Container.terminationMessagePath` - this has some limitations in that Windows doesn't support mapping single files. The default value is `/dev/termination-log`, which does work because it does not exist on Windows by default.
+
+
+`V1.Container.volumeMounts`
+
+> TODO: check if mounting different volumes to containers in the same pod works. May need to go on test TODO list
+
+
+### V1.Pod
+
+> TODO: double check other fields in [source](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L2743)
+
+
+`V1.Pod.hostIPC`, `v1.pod.hostpid` - host namespace sharing is not possible on Windows
+
+`V1.Pod.hostNetwork` - There is no Windows OS support to share the host network
+
+`V1.Pod.dnsPolicy` - ClusterFirstWithHostNet - is not supported because Host Networking is not supported on Windows.
+
+`V1.podSecurityContext.runAsUser` provides a UID, not available on Windows
+`V1.podSecurityContext.supplementalGroups` provides GID, not available on Windows
+
+`V1.Pod.shareProcessNamespace` - this is an alpha feature, and depends on Linux cgroups which are not implemented on Windows
+
+`V1.Pod.volumeDevices` - this is an alpha feature, and is not implemented on Windows. Windows cannot attach raw block devices to pods.
+
+`V1.Pod.Volumes` - EmptyDir, Secret, ConfigMap, HostPath - all work and have tests in TestGrid
```

Comment:
Does medium = Memory work for emptyDir?

Reply:
no, added a note on that. Will be in next commit.